packages/@aws-cdk/aws-iam/lib/policy-document.ts

```diff
@@ -90,12 +91,16 @@ export class AccountPrincipal extends ArnPrincipal {
  * An IAM principal that represents an AWS service (i.e. sqs.amazonaws.com).
  */
 export class ServicePrincipal extends PolicyPrincipal {
-  constructor(public readonly service: string) {
+  constructor(public readonly service: string, private readonly region?: string) {
     super();
   }
 
   public policyFragment(): PrincipalPolicyFragment {
-    return new PrincipalPolicyFragment({ Service: [ this.service ] });
+    const factName = Facts.servicePrincipal(this.service);
+    const principal = this.region != null
+      ? RegionInfo.find(this.region, factName) || this.service
```

Comment:
...and then I would expect this to be called `Fact.find(region, factName)`.

Reply:
Interesting... I liked it called "RegionInfo" something...

Reply:
I don't mind `RegionInfo`, but ideally I'd like the terms to line up.

* A fact *name* is not the same as a *fact*.

Right now the `find` operation basically says: 

"Given a fact name, I will get you region info". 

To me it seems more logical to say:

"Given a fact name, I will get you a fact".

It's okay to not change it. I'm just telling what I would expect given those terms.